src/lib/y2storage/bcache.rb

```diff
@@ -59,6 +64,21 @@ def inspect
       "<Bcache #{name} #{bcache_cset.inspect} -> #{blk_device}>"
     end
 
+    def self.usable_blk_devices(device_graph)
+      device_graph.blk_devices.select do |dev|
+        dev.component_of.empty? &&
+          (dev.filesystem.nil? || dev.filesystem.mount_point.nil?) &&
+          (!dev.respond_to?(:partitions) || dev.partitions.empty?)
+      end
+    end
+
+    # Device name suitable for newly created bcache
```

Comment:
In other classes, this logic is built into the library itself. Like `Md.find_free_numeric_value` or the name of `PartitionTables::Base.unused_partition_slots`.

Reply:
so should I modify library? I do not see similar method in library for now: https://github.com/openSUSE/libstorage-ng/blob/master/storage/Devices/Bcache.h
I tried to avoid it if possible as ruby is faster to write something and easier for testing :)

Reply:
Take into account that bcache support in the library is in Hackweek state, so don't expect to see everything there.

Of course, I also prefer to read and test Ruby code, but for consistency with other parts in which available numbers and/or names are calculated, I honestly believe this should go to the library.

Sorry dude. :wink: